community/logging/src/main/java/org/neo4j/logging/async/AsyncLogEvent.java

```diff
@@ -153,32 +152,20 @@ public String toString()
         return super.toString();
     }
 
-    private String timestamp()
-    {
-        return TIMESTAMP.format( timestamp );
-    }
-
     private abstract static class BulkLogger
     {
         abstract void process( long timestamp, Object target );
     }
 
-    private static final ThreadLocalFormat TIMESTAMP = new ThreadLocalFormat();
-
-    private static class ThreadLocalFormat extends ThreadLocal<DateFormat>
+    private String timestamp()
     {
-        String format( long timestamp )
-        {
-            return get().format( new Date( timestamp ) );
-        }
-
-        @Override
-        protected DateFormat initialValue()
-        {
-            SimpleDateFormat format = new SimpleDateFormat( "yyyy-MM-dd HH:mm:ss.SSSZ" );
-            format.setTimeZone( TimeZone.getTimeZone( "UTC" ) );
-            return format;
-        }
+        return DATE_FORMAT_THREAD_LOCAL.get().format( new Date( timestamp ) );
     }
+
+    private static final ThreadLocal<DateFormat> DATE_FORMAT_THREAD_LOCAL = ThreadLocal.withInitial( () -> {
```

Comment:
This is very similar to the `org.neo4j.helpers.Format.date` family of methods.

Reply:
Sadly that resides in kernel and is not accessible in the logging module, and it's public so can't move it... :(